### PR TITLE
Keep Babies Safe #71

### PR DESCRIPTION
**Pull request by prajwal-144**
# Pull Request for DL-Simplified 💡

## Issue Title : HackerEarth Keep Babies Safe

- **To classify products into consumer_products and toys and to tag them as per their brand names.** : 
- **Name:** Prajwal Uday
- **GitHub ID:** prajwal-144
- **Email ID:** uprajwal20@gmail.com
- **Idenitfy yourself:** SSOC Participant

Closes: #71 

### Describe the add-ons or changes you've made 📃

The code uses Resnet50, a pre-trained CNN, for feature extraction, we remove its topmost/head or the final layer of neurons used for prediction of classes, we then feed our image to the CNN and get a feature vector as an output, which essentially is a flattened array of all the feature maps learned by our CNN at the second last layer of Resnet50. This output vector can be fed into any clustering algorithm ( kmeans(n_cluster = 2) or agglomerative clustering) which classifies our images into the desired number of classes.

## Type of change ☑️

What sort of change have you made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested? ⚙️

OCR libraries were used to perform the task.
Tesseract OCR and Keras OCR was used.

## Checklist: ☑️

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added things that prove my fix is effective or that my feature works.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Review by abhisheks008 — CHANGES_REQUESTED** [review]
Everything looks good, still few changes are required.
1. In the `.ipynb` and `README.md` file, mention the accuracy scores of the models those have been developed. Using that accuracy score, conclude the README.md file, and put the better fitted model.
2. The project folder name should be **HackerRank Keep Babies Safe**.

@prajwal-144

**Comment by prajwal-144** [comment]
I looked into evaluating an ocr model, specifically tesseract. It needs a ground truth reference text so that it can compare the result to the reference. But since this model is extracting text from image from a dataset of 1130 images, i'm not sure if that can be done in this case. 
@abhisheks008

**Comment by abhisheks008** [comment]
What are the other models you are familiar with? Try to use them, coz with a single model we can't compare the accuracy scores. @prajwal-144

**Comment by abhisheks008** [comment]
@prajwal-144 sorry for bothering you again. You have to change the folder name as, **HackerRank Keep Babies Safe** instead of `HackerRank_KeepBabiesSafe`.

**Review by abhisheks008 — APPROVED** [review]
You already have done a great work. But after updating the project with new models and comparing them altogether makes your project more stand out than others. Keep going!

@prajwal-144 
Your PR is approved and ready to be merged.
